homeassistant/components/evohome.py

```diff
@@ -0,0 +1,1941 @@
+            self._available = False
+            debug_code = '0x02'
+
+        elif self._status and (self._type & EVO_SLAVE):
+            # (un)available because (web site via) client api says so
+            self._available = \
+                bool(self._status['temperatureStatus']['isAvailable'])
+            debug_code = '0x03'
+
+        else:  # is available
+            self._available = True
+
+        if not self._available and \
+                self._timers['statusUpdated'] != datetime.min:
+            # this isn't the first (un)available (i.e. is after 1st update())
+            _LOGGER.warning(
+                "available(%s) = %s, debug code = %s, self._status = %s",
+                self._id,
+                self._available,
+                debug_code,
+                self._status
+            )
+
+#       _LOGGER.debug("available(%s) = %s", self._id, self._available)
+        return self._available
+
+    @property
+    def should_poll(self):
+        """Only the Controller will ever be polled.
+
+        The Controller is usually (but not always) polled, and it will obtain
+        the state data for its slaves (which themselves are never polled).
+        """
+        if self._type & EVO_MASTER:
+            pass  # the Controller will decide this as it goes along
+        else:
+            self._should_poll = False
+
+#       _LOGGER.debug("should_poll(%s) = %s", self._id, self._should_poll)
+        return self._should_poll
+
+    @property
+    def supported_features(self):
+        """Get the list of supported features of the Controller."""
+# It will likely be the case we need to support Away/Eco/Off modes in the HA
+# fashion, even though evohome's implementation of these modes are subtly
+# different - this will allow tight integration with the HA landscape e.g.
+# Alexa/Google integration
+#       feats = self._supported_features
+#       _LOGGER.debug("supported_features(%s) = %s", self._id, feats)
+        return self._supported_features
+
+    @property
+    def operation_list(self):
+        """Return the list of available operations.
+
+        Note that, for evohome, the operating mode is determined by - but not
+        equivalent to - the last operation (from the operation list).
+        """
+#       _LOGGER.debug("operation_list(%s) = %s", self._id, self._op_list)
+        return self._op_list
+
+    @property
+    def current_operation(self):
+        """Return the operation mode of the evohome entity."""
+        if self._type & EVO_MASTER:
+            curr_op = self._status['systemModeStatus']['mode']
+        elif self._type & EVO_ZONE:
+            curr_op = self._status[SETPOINT_STATE]['setpointMode']
+        elif self._type & EVO_DHW:
+            curr_op = self._status['stateStatus']['mode']
+
+#       _LOGGER.debug("current_operation(%s) = %s", self._id, curr_op)
+        return curr_op
+
+    @property
+    def state_attributes(self):
```

Comment:
You are not allowed to override `state_attributes`. You can override `device_state_attributes`

Reply:
I'd had this bit in for some deep-level debugging - not needed now, so has been taken out with latest commit.